packages/caliper-core/test/worker/tx-observers/internal-tx-observer.js

```diff
@@ -0,0 +1,296 @@
+     */
+    static resolvePath(path) {
+        return 'fake/path';
+    }
+
+    /**
+     * Mocked version of the sleep method.
+     * Simulates a sleep by immediately resolving the promise.
+     *
+     * @param {number} ms - The number of milliseconds to sleep (ignored in the mock).
+     * @returns {Promise<void>} - A resolved promise, simulating the sleep.
+     */
+    static sleep(ms) {
+        return Promise.resolve();
+    }
+
+    /**
+     *
+     * @param {*} yaml res
+     * @return {string} the fake yaml
+     */
+    static parseYaml(yaml) {
+        return 'yaml';
+    }
+
+    /**
+     * Mocked version of the getLogger method.
+     * Returns a logger object with warn and error stubs for logging testing.
+     *
+     * @returns {{warn: function, error: function}} - A mocked logger with stubs for `warn` and `error` methods.
+     */
+    static getLogger() {
+        return {
+            warn: warnLogger,
+            error: errorLogger
+        };
+    }
+}
+
+
+mockery.enable({
+    warnOnReplace: false,
+    warnOnUnregistered: false
+});
+mockery.registerMock('../../common/utils/caliper-utils', CaliperUtils);
+
+describe('When using Internal Transaction Observer', () => {
+    const TxUpdateMessage = require('../../../lib/common/messages/txUpdateMessage');
+    const InternalTxObserver = require('../../../lib/worker/tx-observers/internal-tx-observer');
+    const TxResetMessage = require('../../../lib/common/messages/txResetMessage');
+    const ConfigUtil = require('../../../lib/common/config/config-util');
+
+
+    let messengerMock, managerUuid, workerIndex;
+    let clock;
+    const DEFAULT_UPDATE_INTERVAL = 1000; // in milliseconds
+
+    beforeEach(() => {
+        // Mocking dependencies
+        messengerMock = {
+            getUUID: sinon.stub().returns('mocked-messenger-uuid'),
+            send: sinon.stub().resolves()
+        };
+        managerUuid = 'mocked-manager-uuid';
+        workerIndex = 0;
+        clock = sinon.useFakeTimers();
+        warnLogger.reset();
+        errorLogger.reset();
+    });
+
+    afterEach(() => {
+        sinon.restore();
+        mockery.deregisterAll();
+        mockery.disable();
+        clock.restore();
+    });
+
+    describe('Construction', () => {
+        it('should initialize with correct properties', () => {
+            sinon.stub(ConfigUtil, 'get').returns(DEFAULT_UPDATE_INTERVAL);
+            const observer = new InternalTxObserver(messengerMock, managerUuid, workerIndex);
+
+            expect(observer.messengerUUID).to.equal('mocked-messenger-uuid');
+            expect(observer.managerUuid).to.equal(managerUuid);
+            expect(observer.updateInterval).to.equal(DEFAULT_UPDATE_INTERVAL);
+            expect(observer.intervalObject).to.be.undefined;
+        });
+
+        it('should log warning if interval is invalid and use default value', async () => {
+            sinon.stub(ConfigUtil, 'get').returns(-1); // Invalid interval
```

Comment:
We should change this to
```
sinon.stub(ConfigUtil, 'get').withArgs(ConfigUtil.keys.Observer.Internal.Interval).returns(-1)
```
just to be explicit about which property we are stubbing